### PR TITLE
Use Error property when throwing exception

### LINKED ISSUE #1376: [BUG] Unable to throw a ValidationApiException exception
I have a Web API that returns a `ProblemDetails` with the `application/problem+json` content type:

image

Then, I have the following Refit interface:

```
public interface IPeopleServiceApi
{
    [Post("/people")]
    Task<ApiResponse<Person>> GetAsync(Person person);
}
```

If I try to call this API from Refit, when there is a validation problem, in the `Error` property of the `ApiResponse` I see a `ValidationApiException` object:

image

However, if I call `EnsureSuccessStatusCodeAsync`, the exception that is raised is `ApiException`, not `ValidationApiException`. Moreover, the `HasContent` property of the exception is _false_:

image

If, instead, I change the interface to directly return a `Person `instead of an `ApiResponse<Person>`, when I call the API, a `ValidationApiException` is thrown, with the `Content `correctly set.

**Steps To Reproduce**

1. Invoke an API that return a problem details with the `application/problem+json` content type
2. Define a method in the interface for Refit that returns an _ApiReponse<T>_ for this API
3. Invoke the API from the client
4. Call the `ApiResponse.EnsureSuccessStatusCodeAsync` method
5. An `ApiException` is thrown

**Expected behavior**
A `ValidationApiException` should be thrown

**Environment**
- OS:       Windows 11
- Device:   Desktop
- Version:   6.3.2

### PR DESCRIPTION
**Pull request by marcominerva**
**What kind of change does this PR introduce?**
Bug fix

**What is the current behavior?**
See https://github.com/reactiveui/refit/issues/1376

**Please check if the PR fulfills these requirements**
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

**Other information**:
Fixes #1376

### CODE REVIEW

**Comment by 304NotModified** [comment]
> ✅️ Tests for the changes have been added (for bug fixes / features)

I dont see any tests in this PR?

**Comment by marcominerva** [comment]
Is adding a new test class to Refit.Tests enough? Thank you @304NotModified

**Comment by 304NotModified** [comment]
That's a bit hard to decide without any test code. 

But I think only an unit tests is needed, not a integration test / service test or something like that. (See https://martinfowler.com/articles/practical-test-pyramid.html)

**Comment by marcominerva** [comment]
Thank you for the clarification @304NotModified, I have added a Unit test for this change.

File: Refit.Tests/ResponseTests.cs [hunk 1]
```diff
@@ -116,6 +116,43 @@ public async Task ThrowsValidationException()
...
+            var expectedResponse = new HttpResponseMessage(HttpStatusCode.BadRequest)
+            {
+                Content = new StringContent(JsonConvert.SerializeObject(expectedContent))
+            };
+
+            expectedResponse.Content.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("application/problem+json");
+            mockHandler.Expect(HttpMethod.Get, "http://api/GetApiResponseTestObject")
+                .Respond(req => expectedResponse);
+
+            try
```
Comment (304NotModified) [inline Refit.Tests/ResponseTests.cs:145]: I think you should use `Assert.ThrowsAsync...` - see other tests

File: Refit.Tests/ResponseTests.cs [hunk 2]
```diff
@@ -116,6 +116,43 @@ public async Task ThrowsValidationException()
...
+            {
+                using var response = await fixture.GetApiResponseTestObject();
+                await response.EnsureSuccessStatusCodeAsync();
+            }
+            catch (Exception ex)
+            {
+                Assert.True(ex is ValidationApiException);
+                Assert.NotNull((ex as ValidationApiException)?.Content);
```
Comment (304NotModified) [inline Refit.Tests/ResponseTests.cs:153]: I would expect that we could do better here, right now it's only checking if the content is `not null`

**Comment by 304NotModified** [comment]
Thanks for the tests! Added some comments to make them a bit better :)

**Comment by marcominerva** [comment]
Done :) Let me know if now is OK!

**Review by 304NotModified — APPROVED** [review]
LGTM

**Comment by nathanpovo** [comment]
This will also fix #1465

**Pull request merged**